### PR TITLE
Add HDF5 file to datafilehub

### PR DESCRIPTION
**Pull request by lingyun1010**
Manual checks

- [ ] This branch is passing the CI tests as part of a pull request in the atlas-web-bulk or atlas-web-scxa repos and I have indicated that PR here.

### CODE REVIEW

File: src/test/java/uk/ac/ebi/atlas/model/experiment/ExperimentBuilder.java [hunk 1]
```diff
@@ -51,11 +51,13 @@
     private final int dataProvidersSize = RNG.nextInt(3);
     private final int alternativeViewsSize = RNG.nextInt(4);
 
-    List<String> technologyType = Arrays.asList(randomAlphabetic(6));
+    ImmutableList<String> technologyType = ImmutableList.copyOf(Arrays.asList(randomAlphabetic(6)));
```
Comment (upendrakumbham) [inline src/test/java/uk/ac/ebi/atlas/model/experiment/ExperimentBuilder.java:54]: Use `'var'`

File: src/test/java/uk/ac/ebi/atlas/model/experiment/ExperimentBuilder.java [hunk 2]
```diff
@@ -51,11 +51,13 @@
...
     ExperimentType experimentType = getRandomExperimentType();
     String experimentAccession = generateRandomExperimentAccession();
-    ImmutableSet<String> secondaryExperimentAccessions = RNG.nextBoolean() ? ImmutableSet.of(generateRandomPrideExperimentAccession())
-            : ImmutableSet.of();
+    ImmutableSet<String> secondaryExperimentAccessions =
```
Comment (upendrakumbham) [inline src/test/java/uk/ac/ebi/atlas/model/experiment/ExperimentBuilder.java:57]: Replace with `'var'`

File: src/main/java/uk/ac/ebi/atlas/resource/DataFileHub.java [hunk 3]
```diff
@@ -497,6 +499,10 @@ public SingleCellExperimentFiles getSingleCellExperimentFiles(String experimentA
                                             SINGLE_CELL_MARKER_GENES_FILE_PATH_TEMPLATE,
                                             experimentAccession,
                                             k.toString())));
+            projectHdf5 = new TsvFile.ReadOnly(
+                    experimentsMageTabDirLocation,
+                    SINGLE_CELL_HDF5_PATH_TEMPLATE,
```
Comment (upendrakumbham) [inline src/main/java/uk/ac/ebi/atlas/resource/DataFileHub.java:504]: can we add a test case @lingyun1010 for this new feature in the `DataFileHubIT.java`?

File: src/test/java/uk/ac/ebi/atlas/model/experiment/ExperimentBuilder.java (hunk 1 above)
Comment (lingyun1010) [reply src/test/java/uk/ac/ebi/atlas/model/experiment/ExperimentBuilder.java:54]: This was the @alfonsomunozpomer commit to resolve the Jenkins test.

File: src/test/java/uk/ac/ebi/atlas/model/experiment/ExperimentBuilder.java (hunk 2 above)
Comment (lingyun1010) [reply src/test/java/uk/ac/ebi/atlas/model/experiment/ExperimentBuilder.java:57]: Same here.

File: src/main/java/uk/ac/ebi/atlas/resource/DataFileHub.java (hunk 3 above)
Comment (lingyun1010) [reply src/main/java/uk/ac/ebi/atlas/resource/DataFileHub.java:504]: We don't really have HDF5 file in the existing test experiments. And for some reasons, this `DataFileHubIT.java` only test for `gxa` experiments.

**Review by upendrakumbham — APPROVED** [review]
All looks good to me. Please merge away.

**Pull request merged**